lib/gmail-adapter.js

```diff
@@ -16,42 +17,87 @@ var Read = Juttle.proc.base.extend({
         this.logger.debug('intitialize', options, params);
         this.gmail = google.gmail('v1');
 
-        var allowed_options = ['raw', 'from', 'to'];
+        var time_related_options = ['from', 'to', 'last'];
+        var allowed_options = time_related_options.concat(['raw']);
         var unknown = _.difference(_.keys(options), allowed_options);
 
         if (unknown.length > 0) {
-            throw new Error('Unknown option ' + unknown[0]);
+            throw JuttleErrors.syntaxError('RT-UNKNOWN-OPTION-ERROR',
+                                           {proc: 'gmail', option: unknown[0], location: location});
+        }
+
+        // One of 'from', 'to', or 'last' must be present.
+        var opts = _.intersection(_.keys(options), time_related_options);
+        if (opts.length === 0) {
+            // Waiting on a juttle release that incorporates https://github.com/juttle/juttle/pull/108
+            //throw JuttleErrors.syntaxError('RT-MISSING-TIMERANGE-ERROR', {location: location});
+            throw new Error('Error: One of -from, -to, or -last must be specified to define a query time range');
+        }
+
+        // If 'from'/'to' are present, 'last' can not be present.
+        if ((_.has(options, 'from') || _.has(options, 'to')) &&
+            _.has(options, 'last')) {
+            throw JuttleErrors.syntaxError('RT-LAST-FROM-TO-ERROR', {location: location});
+        }
+
+        // 'from' must be before 'to'
+        if (_.has(options, 'from') && _.has(options, 'to') &&
+            options.from > options.to) {
+            throw JuttleErrors.syntaxError('RT-TO-FROM-MOMENT-ERROR', {location: location});
+        }
+
+        // If 'last' is specified, set appropriate 'from'/'to'
+        if (_.has(options, 'last')) {
+            this.from = JuttleMoment.now().subtract(options.last);
```

Comment:
I believe the right value of _now_ to use is `this.program.now` so that everything in the running program is synced to the same value.